gotrue/types.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import annotations
+
+from enum import Enum
+from time import time
+from typing import Any, Callable, Dict, List, Optional, Type, TypeVar, Union
+
+from httpx import Response
+from pydantic import BaseModel, root_validator
+
+from gotrue.helpers import check_response
+
+T = TypeVar("T", bound=BaseModel)
+
+
+def determine_session_or_user_model_from_response(
+    response: Response,
+) -> Union[Type[Session], Type[User]]:
+    if "access_token" in response.json():
+        return Session
+    return User
+
+
+class BaseModelFromResponse(BaseModel):
+    @classmethod
+    def parse_response(cls: Type[T], response: Response) -> T:
+        check_response(response)
+        return cls.parse_obj(response.json())
+
+
+class CookieOptions(BaseModelFromResponse):
+    name: str
+    """The name of the cookie. Defaults to `sb:token`."""
+    lifetime: int
+    """The cookie lifetime (expiration) in seconds. Set to 8 hours by default."""
+    domain: str
+    """The cookie domain this should run on.
+    Leave it blank to restrict it to your domain."""
+    path: str
+    same_site: str
+    """SameSite configuration for the session cookie.
+    Defaults to 'lax', but can be changed to 'strict' or 'none'.
+    Set it to false if you want to disable the SameSite setting."""
+
+
+class Identity(BaseModelFromResponse):
+    id: str
+    user_id: str
+    provider: str
+    created_at: str
+    updated_at: str
+    identity_data: Optional[Dict[str, Any]] = None
+    last_sign_in_at: Optional[str] = None
+
+
+class User(BaseModelFromResponse):
+    app_metadata: Dict[str, Any]
+    aud: str
```

Comment:
What data does this field contain? Maybe adding some documentation for the fields on this class might be helpful for users, just a thought

Reply:
Where can we find a doc for that? Neither `gotrue` nor `gotrue-js` has.

Reply:
Oh, I don't know either which is why I asked 🙂 Having fields that no one knows the functionality of just feels iffy to me

Reply:
Pydantic fields can be used to add a description of the type.

Reply:
According to this https://github.com/supabase/gotrue/blob/1dddc248fb820e638f0d54b9d51c2803295b5c1d/models/user.go#L355 is to represent the user's audience.

This may also be related to the `GOTRUE_JWT_AUD` parameter of https://supabase.io/docs/gotrue/server/about#json-web-tokens-jwt

On the other hand, I made a query in an example project and an example user and I got this

![image](https://user-images.githubusercontent.com/36774373/141693573-d04dfc91-72bd-4c90-9fff-1550b56abc05.png)

### My proposals for description are:

1. The user's audience.
2. The user's audience, use audiences to group users.

Reply:
I would use a period:
The user's audience. Use audiences to group users.